tests/PHPStan/Levels/data/stringOffsetAccess-7.json

```diff
@@ -9,11 +14,21 @@
         "line": 31,
         "ignorable": true
     },
+    {
+        "message": "Offset int|object might not exist on array|string.",
+        "line": 35,
+        "ignorable": true
+    },
     {
         "message": "Possibly invalid array key type int|object.",
         "line": 35,
         "ignorable": true
     },
+    {
+        "message": "Offset int|object might not exist on array{baz: 21}|array{foo: 17, bar: 19}.",
```

Comment:
This looks wrong for `array{foo: 17, bar: 19}|array{baz: 21}`. `int|object` never exists on these arrays.

Reply:
Indeed, I think this is because of https://github.com/phpstan/phpstan-src/pull/4272 which need to be fix/merged first then.